store/src/java/com/zimbra/cs/service/mail/GetCalendarItem.java

```diff
@@ -88,26 +89,23 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         if (sync) {
             fields |= Change.CONFLICT;
         }
-        Element response = getResponseElement(zsc);
-        CalendarItem calItem = null;
-        mbox.lock.lock(false);
-        try {
-            if (uid != null) {
-                calItem = mbox.getCalendarItemByUid(octxt, uid);
-                if (calItem == null) {
-                    throw MailServiceException.NO_SUCH_CALITEM(uid);
-                }
-            } else {
-                calItem = mbox.getCalendarItemById(octxt, iid);
-                if (calItem == null) {
-                    throw MailServiceException.NO_SUCH_CALITEM(iid.toString());
-                }
-            }
-        } finally {
-            mbox.lock.release();
-        }
-        ToXML.encodeCalendarItemSummary(response, ifmt, octxt, calItem, fields, includeInvites, includeContent);
-
-        return response;
+		Element response = getResponseElement(zsc);
+		CalendarItem calItem = null;
+		try (final MailboxLock l = mbox.lockFactory.readLock()) {
+			l.lock();
+			if (uid != null) {
+				calItem = mbox.getCalendarItemByUid(octxt, uid);
+				if (calItem == null) {
+					throw MailServiceException.NO_SUCH_CALITEM(uid);
+				}
+			} else {
+				calItem = mbox.getCalendarItemById(octxt, iid);
+				if (calItem == null) {
+					throw MailServiceException.NO_SUCH_CALITEM(iid.toString());
+				}
+			}
+			ToXML.encodeCalendarItemSummary(response, ifmt, octxt, calItem, fields, includeInvites, includeContent);
```

Comment:
The order changed here.  It's holding the lock while calling encodeCalendarItemSummary.

Reply:
moved to another branch and recreated your comment there.